source/validate_instruction.cpp

```diff
@@ -176,6 +176,31 @@ spv_result_t LimitCheckSwitch(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+// Ensure the number of variables of the given class does not exceed the limit.
+spv_result_t LimitCheckNumVars(ValidationState_t& _,
+                               const SpvStorageClass storage_class) {
+  if (SpvStorageClassFunction == storage_class) {
```

Comment:
This is fine but it somewhat duplicates the logic in the callee function that separates out Function storage class variables from non-Function storage class.